### PR TITLE
add HeapSort-code

### PR DESCRIPTION
**Pull request by ccm18454090438**

### CODE REVIEW

**Comment by ccm18454090438** [comment]
@UDLD    add HS.java -code

File: java-code/HeapSort.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import java.util.Scanner;
+
+public class HeapSort {
+    public static void swap(int[] a,int b,int c)
+    {
+        int d=a[b];
+        a[b]=a[c];
+        a[c]=d;
+    }
+    public static void bbll(int[] a,int n)
```
Comment (UDLD) [inline java-code/HeapSort.java]: the method name is bad

File: java-code/HeapSort.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                if((n>=2*(2*i+2)+1+1 && a[2*(2*i+2)+1]>a[2*i+2] ) || (n>=2*(2*i+2)+2+1 && a[2*(2*i+2)+2]>a[2*i+2]))
+                    bbll(a,n);
+            }
+        }
+
+    }
+    public static  void main(String[] arge) {
+        Scanner in = new Scanner(System.in);
+        int k=7;
+        int[] a =new int[k];
```
Comment (UDLD) [inline java-code/HeapSort.java]: not support other length test

File: java-code/HeapSort.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import java.util.Scanner;
```
Comment (UDLD) [inline java-code/HeapSort.java]: code style bad

File: java-code/HeapSort.java (hunk 1 above)
Comment (ccm18454090438) [reply java-code/HeapSort.java]: @UDLD Modified please pass

**Review by UDLD — APPROVED** [review]
good

**Pull request merged**